### PR TITLE
Use correct Exponent type

### PR DESCRIPTION
**Pull request by xantares**

### CODE REVIEW

File: Number_types/include/CGAL/GMP/Gmpzf_type.h [hunk 1]
```diff
@@ -422,7 +422,7 @@ std::pair<std::pair<double, double>, long> Gmpzf::to_interval_exp() const
   // get surrounding interval of the form [l * 2 ^ k, u * 2^ k]
   // first get mantissa in the form l*2^k, with 0.5 <= d < 1;
   // truncation is guaranteed to go towards zero
-  long k = 0;
+  Exponent k = 0;
   double l = mpz_get_d_2exp (&k, man());
```
Comment (mglisse) [inline Number_types/include/CGAL/GMP/Gmpzf_type.h]: Well, the correct type for k is whatever `mpz_get_d_2exp` is documented to take as first argument (minus the pointer). According to the GMP documentation, `long` is right, and the reverse patch in function `to_double` would make sense.

Comment (xantares) [inline Number_types/include/CGAL/GMP/Gmpzf_type.h]: it's for mpir, yes, it helps if it's a typedef

Comment (mglisse) [inline Number_types/include/CGAL/GMP/Gmpzf_type.h]: https://github.com/wbhart/mpir/blob/master/gmp-h.in#L918 seems to say that `mpz_get_d_2exp` also expects a `long` in MPIR now. If we change things to help older versions of MPIR, that's going to break future ones...

**Comment by mglisse** [comment]
I am ok with this patch for consistency reasons. If some day we bother to improve things, we can change the 3 instances at the same time.

**Comment by xantares** [comment]
exactly

**Pull request merged**